ext/fg/js/frontend.js

```diff
@@ -26,24 +26,24 @@
  */
 
 class Frontend extends TextScanner {
-    constructor(popup) {
+    constructor(popup, getUrl) {
```

Comment:
`getUrl` could be made to accept `null` also, and just return `window.location.href` in that case. That would remove an `await` call for the typical use case (non-nested popup on a content page).